scrapy/extensions/feedexport.py

```diff
@@ -341,14 +345,16 @@ def close_spider(self, spider):
         return defer.DeferredList(deferred_list) if deferred_list else None
 
     def _close_slot(self, slot, spider):
-        if not slot.itemcount and not slot.store_empty:
+        if not self.batches[slot.uri_template].updated_once and not slot.store_empty:
```

Comment:
Would `current_item_count` work instead of `updated_once`? (and remove the need to define `updated_once` in the `Batch` class)

Reply:
Since the goal is to provide more batch triggers than just item count, I had thought of this so that when the file size and time duration are introduced as well or maybe a custom batch class is being used, `updated_once` can be used to determine if a batch is empty.